### PR TITLE
[RFC][WIP][DO NOT MERGE!] rootless networking

### PR DESCRIPTION
**Pull request by giuseppe**
not ready to be used, this is to start a discussion around an usermode network configuration.

It is based on this tool: https://github.com/giuseppe/slirp-forwarder

It internally uses libslirp, we might want to use the version from qemu, but that requires more work.

Also I discovered there is already another attempt at creating a slirp tool for rootless networks: https://github.com/AkihiroSuda/slirp4netns and we can probably collaborate together for it.

this is how a container looks like, there is nothing requiring root privileges or suid applications to configure the network:

``console
$ bin/podman run --rm -ti alpine ifconfig -a
lo        Link encap:Local Loopback  
          LOOPBACK  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

tap0      Link encap:Ethernet  HWaddr 36:FF:87:4E:54:26  
          inet6 addr: fe80::34ff:87ff:fe4e:5426/64 Scope:Link
          UP BROADCAST RUNNING  MTU:1500  Metric:1
          RX packets:2 errors:0 dropped:0 overruns:0 frame:0
          TX packets:6 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1180 (1.1 KiB)  TX bytes:992 (992.0 B)

$ pkill dhclient; pkill slirp-forwarder; rm /tmp/net # delete any leftover
```

### CODE REVIEW

File: libpod/container.go [hunk 1]
```diff
@@ -236,6 +236,7 @@ type ContainerConfig struct {
 	UserNsCtr   string `json:"userNsCtr,omitempty"`
 	UTSNsCtr    string `json:"utsNsCtr,omitempty"`
 	CgroupNsCtr string `json:"cgroupNsCtr,omitempty"`
+	NetNs       string `json:"netNs,omitempty"`
```
Comment (mheon) [inline libpod/container.go]: Why do we need this? Why not create it in cmd/podman and pass it in the spec?

Comment (giuseppe) [reply libpod/container.go]: I've not tried it but would it not get lost once we restart the container?

**Comment by mheon** [comment]
Commit 1 looks fine. Commit 2 is probably better off happening entirely in `cmd/podman` and directly changing the spec we pass into libpod.

File: libpod/container.go (hunk 1 above)
Comment (mheon) [reply libpod/container.go]: Should be fine, we retain the whole spec and the path is in the spec?

Comment (giuseppe) [reply libpod/container.go]: yes it is in the spec, but we delete the OCI container when we do a `podman restart`

**Comment by giuseppe** [comment]
I'm closing this PR in favor of https://github.com/projectatomic/libpod/pull/1145

The other experimental change will be added later and it'll probably use https://github.com/AkihiroSuda/slirp4netns

**Pull request closed without merging**